build/script.js

```diff
@@ -102,6 +104,34 @@ gulp.task('scripts-watch', createScriptsStream(false));
  * 'async'.
  */
 function createCompileTask(translation) {
+
+  let outputDir =
+      translation ? path.join(conf.paths.i18nProd, `/${translation.key}`) : conf.paths.prodTmp;
+
+  return (
+      (next) =>
+          gulp.src([
+                // Application source files.
+                path.join(conf.paths.frontendSrc, '**/*.js'),
+                // Partials generated by other tasks, e.g., Angular templates.
+                path.join(conf.paths.partials, '**/*.js'),
+                // Include base.js to enable some compiler functions, e.g., @export annotation
+                // handling and getMsg() translations.
+                path.join(
+                    conf.paths.bowerComponents, 'google-closure-library/closure/goog/base.js'),
+              ])
+              .pipe(closureCompiler(translation))
+              .pipe(gulpIf('**/app.js',gulpReplace('BUILD_DASHBOARD_VERSION', conf.deploy.version.release)))
+              .pipe(gulpIf('**/app.js',gulpReplace('BUILD_GIT_COMMIT', process.env.TRAVIS_COMMIT)))
```

Comment:
Did you try pipe before closureCompiler task? gulp.src first takes all the files in frontend src directory + some partial files and pipes them to closure compiler. Closure compiler bundles them to app.js so that's probably why appconfig is not found.

You can always just write some debug function as a pipe and check what files are available in there. There are also gulp-debug, gulp-print modules that can be useful for debugging. Gulp task replace looks promising. You could use that once everything else is working.